### PR TITLE
[EC-386] Azure DevOps pipelines migrated into GitHub Actions

### PR DESCRIPTION
**Pull request by mamu0**
#### List of Changes

The two workflows on Azure DevOps

- `code-review`
- `deploy-pipelines`

have been recreated in the form of GitHub Actions.

They have been inserted in the `.github/workflows` folder.

Minor changes done inside `tslint.json` to permit correct `yarn lint` command execution.

Added `pre-commit` configuration.

Added `infra` folder with `identity` and `repository` infrastructure, and added branch protection on `master` (already applied).

Edited `.gitignore` and `CODEOWNER` files.

#### Motivation and Context

Since all the pipelines on Azure DevOps are gradually being decommissioned, it was decided to recreate them on the different repos such as GitHub Actions.

#### How Has This Been Tested?

Currently they have been tested by inserting

```
push:
  branches:
    - EC-386-devops-decommissioning
```
so as to execute them automatically at each test push.
To do the Deploy test i've used this step only to check if all will be called correctly:
```
      - name: Deploy
        run: |
          az webapp up \
            --dryrun \
            --resource-group ${{ inputs.resource_group_name }} \
            --name ${{ inputs.app_name }} \
```

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/identity/prod/locals.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+locals {
+  prefix    = "io"
+  env_short = "p"
+  env       = "prod"
+  location  = "italynorth"
+  project   = "${local.prefix}-${local.env_short}"
+  domain    = "dev-portal-backend"
```
Comment (Krusty93) [inline infra/identity/prod/locals.tf:7]: I suggest
```suggestion
  domain    = "devportal-backend"
```

To use `devportal` as `domain`, and `backend` a `app_name` (I'm referring to Azure resources)

File: infra/repository/github_repository.tf [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      status = "enabled"
+    }
+
+    # advanced_security {
+    #   status = "enabled"
+    # }
+  }
+}
+
+resource "github_repository_environment" "github_repository_environment_github_pages" {
```
Comment (Krusty93) [inline infra/repository/github_repository.tf:57]: I suggest to create a different file for this environment, as done per `prod-cx` envs

File: infra/repository/locals.tf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    secrets = {
+      "ARM_CLIENT_ID" = data.azurerm_user_assigned_identity.identity_prod_ci.client_id
+    }
+  }
+
+  cd = {
+    secrets = {
+      "ARM_CLIENT_ID" = data.azurerm_user_assigned_identity.identity_prod_cd.client_id
+    }
+    reviewers_teams = ["all-internal-users", "engineering-team-cloud-eng"]
```
Comment (Krusty93) [inline infra/repository/locals.tf:21]:
```suggestion
    reviewers_teams = ["io-platform-green-unit", "engineering-team-cloud-eng"]
```

I think `all-internal-users` should be removed in `eng-github-authorization` repo

File: .github/workflows/deploy-pipelines-prod.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  deploy_on_prod:
+    uses: ./.github/workflows/deploy-pipelines.yaml
+    name: Deploy on PROD
+    secrets: inherit
+    with:
+      environment: 'prod'
+      resource_group_name: 'io-p-selfcare-be-rg'
+      app_name: 'io-p-app-devportal-be'
+      use_staging_slot: true
```
Comment (Krusty93) [inline .github/workflows/deploy-pipelines-prod.yaml:15]: I think this resource doesn't have any slot, and is reachable only from app gateway subnet

**Review by giamma1295 — APPROVED** [review]

**Pull request merged**